src/main/java/network/brightspots/rcv/GuiConfigController.java

```diff
@@ -769,21 +770,22 @@ private boolean checkForSaveAndTabulate() {
     return willContinue;
   }
 
-  private void loadConfig(ContestConfig config) {
-    clearConfig();
+  // clear widgets then set to default values
+  // then populate widgets with values from input config
+  private void loadConfigIntoGui(ContestConfig config) {
+    clearWidgetsAndSetToDefaults();
     RawContestConfig rawConfig = config.getRawConfig();
 
-    OutputSettings outputSettings = rawConfig.outputSettings;
```

Comment:
What motivated you to change everything in this method to read from the ContestConfig instead of the RawContestConfig? @HEdingfield and I debated this a while ago. To reiterate the point I made at the time: the raw config is what we're saving, so shouldn't it be what we use to populate the form fields? This ensures that any normalizing that happens in ContestConfig (e.g. the special logic for maxRankingsAllowed) doesn't corrupt the actual config file.

But perhaps I'm missing the reason that your change to how we handle strings necessitates these changes as well.

Reply:
Yeahhh, shit, ughhh, I dunnooooo........

Reply:
There needs to be logic somewhere to apply the default "" to a null string parameter.  I followed the logic of keep raw raw, and put it in contestConfig which is better than in the UI code.

Reply:
My concern is that if we do it this way, we can't have any special normalization logic in ContestConfig, because every field always has to match the raw version. We already see an example of the problem this creates with maxRankingsAllowed below.

But I don't feel that strongly about it. If you keep it this way, please consider my suggestion for how to change maxRankingsAllowed.

Reply:
Gotcha.  I think we can put the "default" logic into RawConfig and keep the normalization in ContestConfig.  I'll try that and see how it looks.